sabnzbd/interface.py

```diff
@@ -206,15 +207,14 @@ def check_access(access_type: int = 4, warn_user: bool = False) -> bool:
 
     remote_ip = cherrypy.request.remote.ip
 
-    # Check for localhost
-    if is_loopback_addr(remote_ip):
-        return True
+    # Check if the client IP is a loopback address or considered local
+    is_allowed = is_loopback_addr(remote_ip) or is_local_addr(remote_ip)
 
-    if not cfg.local_ranges():
-        # No local ranges defined, allow all private addresses by default
-        is_allowed = is_lan_addr(remote_ip)
-    else:
-        is_allowed = any(ip_in_subnet(remote_ip, r) for r in cfg.local_ranges())
+    # Never check the XFF header unless access would have been granted based on the remote IP alone!
+    if is_allowed and cfg.verify_xff_header() and (xff_header := cherrypy.request.headers.get("X-Forwarded-For")):
+        xff_ips = [ip.strip() for ip in xff_header.split(",")]
+        if not (is_allowed := all(is_local_addr(ip) or is_loopback_addr(ip) for ip in xff_ips)):
```

Comment:
```suggestion
        is_allowed = all(is_local_addr(ip.strip()) or is_loopback_addr(ip.strip()) for ip in xff_header.split(",")))
        if not is_allowed:
```
Or something like that, sometimes the walrus operator makes things harder to read instead of easier (in my opinion :P )